### PR TITLE
Get rid of centroid() in favor of returnCentroid()

### LINKED ISSUE #650: Unify centroid() and returnCentroid()
For historical reasons, we have two of the functions with the same meaning. Can we only have one?

### PR DESCRIPTION
**Pull request by aprokop**
Fix #650.

### CODE REVIEW

**Review by masterleinad — DISMISSED** [review]
If it were a customization point it shouldn't be in a `Details` namespace anyway.

**Comment by aprokop** [comment]
> If it were a customization point it shouldn't be in a `Details` namespace anyway.

Not sure what you mean. A user is allowed to provide his own geometry with `centroid` which does not live in `Details`.

**Comment by masterleinad** [comment]
> Not sure what you mean. A user is allowed to provide his own geometry with `centroid` which does not live in `Details`.

Then, I would rather deprecate in some ways.  Since the existing interface rather uses `returnCentroid()`, it feels better to remove `centroid()` instead.

**Comment by aprokop** [comment]
> Since the existing interface rather uses returnCentroid()

Ah, good point, I should start thinking about what I'm actually doing.

**Comment by aprokop** [comment]
Updated to just using `returnCentroid()`. I think it would be nice to use the `centroid()` name instead, but it's outside of the scope of this PR.

**Comment by aprokop** [comment]
There is 9% slowdown in certain Serial `radius_search_callback`. Not sure if that's real, as `returnCentroid` appears only in sorting the queries.

File: src/geometry/ArborX_DetailsAlgorithms.hpp [hunk 1]
```diff
@@ -417,48 +406,14 @@ struct intersects<PointTag, SphereTag, Point, Sphere>
...
-    c = point;
-  }
-};
-
 template <typename Point>
-struct returnCentroid<PointTag, Point>
+struct centroid<PointTag, Point>
 {
-  KOKKOS_FUNCTION static auto apply(Point const &point) { return point; }
+  KOKKOS_FUNCTION static auto const &apply(Point const &point) { return point; }
```
Comment (dalg24) [inline src/geometry/ArborX_DetailsAlgorithms.hpp:412]: Why did you change to return a reference?

Comment (aprokop) [reply src/geometry/ArborX_DetailsAlgorithms.hpp:412]: Because I don't see a reason to return it by value. Same thing with the centroid of a sphere. I think it was an oversight.

Comment (dalg24) [reply src/geometry/ArborX_DetailsAlgorithms.hpp:412]: Do you see a reason to return by reference?

Comment (aprokop) [reply src/geometry/ArborX_DetailsAlgorithms.hpp:412]: I assume that it makes the compiler job's easier, and makes the intent more clear (in the sense that don't construct anything within the function, rather return something already existing).

**Comment by aprokop** [comment]
It turns out that returning centroid by value is faster than by returning by reference :shrug:

**Review by dalg24 — APPROVED** [review]
